stew/base36.nim

```diff
@@ -0,0 +1,222 @@
+## Copyright (c) 2025 Status Research & Development GmbH
+## Licensed under either of
+##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+## at your option.
+## This file may not be copied, modified, or distributed except according to
+## those terms.
+
+## This module implements Base36 encoding and decoding procedures.
+## This module supports two Base36 alphabets: lowercase `Base36Lc` and uppercase `Base36Uc`.
+
+## This module follows Base36 as defined in https://github.com/multiformats/multibase/blob/f378d3427fe125057facdbac936c4215cc777920/rfcs/Base36.md
+
+{.push raises: [].}
+
+type
+  Base36Status* {.pure.} = enum
+    Error,
+    Success,
+    Incorrect,
+    Overrun
+
+  Base36Alphabet* = object
+    decode*: array[128, int8]
+    encode*: array[36, uint8]
+
+  Base36* = object
+  Base36Uc* = object
+  Base36Lc* = object
+
+  Base36Types* = Base36 | Base36Uc | Base36Lc
+
+  Base36Error* = object of CatchableError
+    ## Base36 specific exception type
+
+func newAlphabet36(s: string): Base36Alphabet =
```

Comment:
It would be useful here to indicate explicitly in the type that it's an `s: static[string]`, i.e. because the intent isn't for this to be used in a random runtime way but as part of a compile-time transformation of one kind of constant to another kind of constant.